File: Fenix-Primavera/Primavera.MotoresPrimavera/Parametros/Instancia.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Primavera.MotoresPrimavera.Parametros
{
    /// <summary>
    /// Classe Instancia para a gestão das versões do primavera e as suas respectivas assemblies
    /// </summary>
    public class Instancia
    {

        public int instancia = 1;
        public string empresa;
        public string usuario;
        public string password;
        public string instanciaSql;
        public string empresaSql;
        public string usuarioSql;
        public string passwordSql;
        public static string versaoErp = "V800";
        public const string pastaConfigV800 = "PRIMAVERA\\SG800";

        public const string pastaConfigV900 = "PRIMAVERA\\SG900";
        public object daConnectionString()
        {
            return "Data Source=" + instanciaSql + ";Initial Catalog= " + empresaSql + ";User Id=" + usuarioSql + ";Password=" + passwordSql;
        }

        public static string daPastaConfig()
        {
            if (versaoErp == "V800")
            {
                return pastaConfigV800;
            }
            else
            {
                return pastaConfigV900;
            }
        }

    }
}

File: CSU_CGA/CSU_CGA/Model/Helper/Instancia.vb
Public Class Instancia
    Public instancia As Integer = 1

    Public empresa As String
    Public usuario As String
    Public password As String
    Public instanciaSql As String
    Public empresaSql As String
    Public usuarioSql As String
    Public passwordSql As String
    Public Shared versaoErp As String = "V800"
    Public Const pastaConfigV800 As String = "PRIMAVERA\\SG800"
    Public Const pastaConfigV900 As String = "PRIMAVERA\\SG900"

    Public Function daConnectionString()
        Return "Data Source=" + instanciaSql + ";Initial Catalog= " + empresaSql + ";User Id=" + usuarioSql + ";Password=" + passwordSql
    End Function

    Public Shared Function daPastaConfig()
        If versaoErp = "V800" Then
            Return pastaConfigV800
        Else
            Return pastaConfigV900
        End If
    End Function

End Class
